scripts/ansible/install-ansible.sh

```diff
@@ -10,6 +10,10 @@ DIR=$(dirname "$0")
 if which yum > /dev/null; then
     yum install git python3-pip -y
 elif which apt-get > /dev/null; then
+    echo "Checking for dpkg locks..."
+    while fuser /var/lib/dpkg/lock > /dev/null 2>&1 || fuser /var/lib/dpkg/lock-frontend > /dev/null 2>&1; do
+        sleep 1
+    done
```

Comment:
There is a subtle issue here. Say the loop terminates and we finish line 16, before line 17 is executed, another process in theory could execute apt, and hold the lock, causing line 17 to fail.
Such situations are common in multi-threaded programs; not sure if it is something we need to guard against in the ansible's use of apt.

Reply:
One approach is to use _flock_ to decide which process gets to use apt. Lock is acquired, apt is used and then lock is released.

Reply:
This is true and it's a good point. The current mitigation for this scenario is that the Jenkins agent that this script runs on only allows one job at a time and then recycled afterwards, so there are no parallel install processes waiting in queue at the same time. Though I like this idea so I will try implementing it and see how it goes.

Reply:
I couldn't solve this issue with flock as apt/dpkg doesn't use it. See https://github.com/openenclave/openenclave/pull/3869#issuecomment-791226643 for more info.